### ISSUE TITLE
When to panic ?

### ISSUE THREAD

**Issue by tarrsalah** [gohttp/response#1]
Should we replace all this lines

```
if err != nil {
        http.Error(res, err.Error(), http.StatusInternalServerError)
        return
}
```

By just `panic(err)` and handle the panic in a separate recovery middleware  ?

**Comment by tj** [comment]
nah I think trying to emulate exceptions bypasses some nice qualities of Go, even if they're super verbose. So far the only lib I've seen in the wild that uses panic/recover a lot was lib/pq but it's super hard to follow what's going on
